### PR TITLE
Added use case 4 and 7 and course entity

### PR DESCRIPTION
**Pull request by agentp88**
This is what I have so far for my use cases

### CODE REVIEW

**Review by nykoadionisio — COMMENTED** [review]
Great work on the code so far! I believe that it follows our CRC model well aside from the slight modifications we discussed in person (which is a good thing!). I do request that we add documentation in the code (JavaDoc) for as many things as you can as it is important we always document our code! Feel free to take a look at my branch (3-use-case-3-register-new-user) for how I wrote out the documentation. Besides that, great work so far!

**Review by HarveyRDonnelly — CHANGES_REQUESTED** [review]
This is really great work Paul ! As mentioned by Nyko, it's a good idea to write your JavaDoc as you go. That said, I'm happy to merge this and you can create a new issue for JavaDoc to be completed later. I suggest however that the UpdateCourseMembership.java class be renamed to UpdateCourseMembershipInteractor for naming consistency. Moreover, we may need to make some changes to account for the StateTracker but we can discuss this at tutorial later today. Again, really great work !

File: src/main/java/entities/Course.java [hunk 1]
```diff
@@ -1,7 +1,68 @@
...
+
+import entities.TestDoc;
+
+/**
+ * Represents a university course
+ */
 public class Course {
+    private final String courseName;
+    private final String courseCode;
+    private final List<TestDoc> tests;
```
Comment (HarveyRDonnelly) [inline src/main/java/entities/Course.java:14]: Will be updated as dictionary in later PR

File: src/main/java/uc/course/updatemembers/UpdateCMemDsGateway.java [hunk 2]
```diff
@@ -0,0 +1,12 @@
+package uc.course.updatemembers;
+
+import java.util.List;
+
+public interface UpdateCMemDsGateway
+{
+    boolean checkIfCourseExists(String courseId);
+    void addCourseEnrollment(String courseId, String userId);
+    void removeCourseEnrollment(String courseId, String userId);
+    List<String> getCourseEnrollment(String userId);
```
Comment (HarveyRDonnelly) [inline src/main/java/uc/course/updatemembers/UpdateCMemDsGateway.java:10]: method name will be updated to getEnrolmentsByUserId for consistency with other use cases

**Review by HarveyRDonnelly — APPROVED** [review]
Great work @agentp88 !! Code is very well written and adheres to the clean architecture. Both use cases are of great quality and documented. Some changes will be needed in a future PR to update the state tracker and make some changes to the course entity

**Pull request merged**